base/repl/LineEdit.jl

```diff
@@ -182,8 +183,10 @@ function _clear_input_area(terminal, state::InputAreaState)
     clear_line(terminal)
 end
 
-prompt_string(s::PromptState) = s.p.prompt
+prompt_string(s::PromptState) = prompt_string(s.p)
+prompt_string(p::Prompt) = prompt_string(p.prompt)
 prompt_string(s::AbstractString) = s
+prompt_string(f::Function) = eval(Expr(:call, f))
```

Comment:
the evals were added here before invokelatest was implemented, that should be the canonical way of doing this now